lms/djangoapps/grades/api/v1/gradebook_views.py

```diff
@@ -230,3 +307,470 @@ def _get_subsections(self, course, graded_only=False):
+                graded_description = '({earned:.2f}/{possible:.2f})'.format(
+                    earned=subsection_grade.graded_total.earned,
+                    possible=subsection_grade.graded_total.possible,
+                )
+                score_earned = subsection_grade.graded_total.earned
+                score_possible = subsection_grade.graded_total.possible
+
+            # TODO: https://openedx.atlassian.net/browse/EDUCATOR-3559 -- Some fields should be renamed, others removed:
+            # 'displayed_value' should maybe be 'description_percent'
+            # 'grade_description' should be 'description_ratio'
+            breakdown.append({
+                'category': subsection_grade.format,
+                'displayed_value': '{:.2f}'.format(subsection_grade.percent_graded),
+                'is_graded': subsection_grade.graded,
+                'grade_description': graded_description,
+                'label': short_label,
+                'letter_grade': course_grade.letter_grade,
+                'module_id': text_type(subsection_grade.location),
+                'percent': subsection_grade.percent_graded,
+                'score_earned': score_earned,
+                'score_possible': score_possible,
+                'subsection_name': subsection_grade.display_name,
+            })
+        return breakdown
+
+    def _gradebook_entry(self, user, course, graded_subsections, course_grade):
+        """
+        Returns a dictionary of course- and subsection-level grade data for
+        a given user in a given course.
+
+        Args:
+            user: A User object.
+            course: A Course Descriptor object.
+            graded_subsections: A list of graded subsections in the given course.
+            course_grade: A CourseGrade object.
+        """
+        user_entry = self._serialize_user_grade(user, course.id, course_grade)
+        breakdown = self._section_breakdown(course, graded_subsections, course_grade)
+
+        user_entry['section_breakdown'] = breakdown
+        user_entry['progress_page_url'] = reverse(
+            'student_progress',
+            kwargs=dict(course_id=text_type(course.id), student_id=user.id)
+        )
+        user_entry['user_id'] = user.id
+        user_entry['full_name'] = user.get_full_name()
+
+        return user_entry
+
+    @verify_course_exists
+    @verify_writable_gradebook_enabled
+    @course_author_access_required
+    def get(self, request, course_key):
+        """
+        Returns a gradebook entry/entries (i.e. both course and subsection-level grade data)
+        for all users enrolled in a course, or a single user enrolled in a course
+        if a `username` parameter is provided.
+
+        Args:
+            request: A Django request object.
+            course_key: The edx course opaque key of a course object.
+        """
+        course = get_course_by_id(course_key, depth=None)
+        self.check_object_permissions(self.request, course)
```

Comment:
What happens if we don't have this line?  Does this do something that `@course_author_access_required` does not?  If so, maybe add a comment (in a later PR).

Reply:
Yeah, this is not needed anymore. It's a relic from what the quest I was on to do the permissions properly.